instrumentation/jaxrs/jaxrs-2.0/src/main/java/io/opentelemetry/auto/instrumentation/jaxrs/v2_0/JaxRsAnnotationsTracer.java

```diff
@@ -18,55 +18,49 @@
 
 import static io.opentelemetry.auto.bootstrap.WeakMap.Provider.newWeakMap;
 
-import io.opentelemetry.OpenTelemetry;
 import io.opentelemetry.auto.bootstrap.WeakMap;
-import io.opentelemetry.auto.bootstrap.instrumentation.decorator.BaseDecorator;
+import io.opentelemetry.auto.bootstrap.instrumentation.decorator.BaseTracer;
 import io.opentelemetry.auto.tooling.ClassHierarchyIterable;
 import io.opentelemetry.trace.Span;
-import io.opentelemetry.trace.Tracer;
 import java.lang.annotation.Annotation;
 import java.lang.reflect.Method;
 import java.util.Map;
 import java.util.concurrent.ConcurrentHashMap;
 import javax.ws.rs.HttpMethod;
 import javax.ws.rs.Path;
 
-public class JaxRsAnnotationsDecorator extends BaseDecorator {
+public class JaxRsAnnotationsTracer extends BaseTracer {
   public static final String ABORT_FILTER_CLASS =
       "io.opentelemetry.auto.instrumentation.jaxrs2.filter.abort.class";
   public static final String ABORT_HANDLED =
       "io.opentelemetry.auto.instrumentation.jaxrs2.filter.abort.handled";
-  public static final String ABORT_PARENT =
-      "io.opentelemetry.auto.instrumentation.jaxrs2.filter.abort.parent";
-  public static final String ABORT_SPAN =
-      "io.opentelemetry.auto.instrumentation.jaxrs2.filter.abort.span";
 
-  public static final JaxRsAnnotationsDecorator DECORATE = new JaxRsAnnotationsDecorator();
-
-  public static final Tracer TRACER =
-      OpenTelemetry.getTracerProvider().get("io.opentelemetry.auto.jaxrs-2.0");
+  public static final JaxRsAnnotationsTracer TRACER = new JaxRsAnnotationsTracer();
 
   private final WeakMap<Class<?>, Map<Method, String>> spanNames = newWeakMap();
 
-  public void onJaxRsSpan(
-      final Span span, final Span parent, final Class target, final Method method) {
-
-    String spanName = getPathSpanName(target, method);
-    updateParent(parent, spanName);
+  public Span startSpan(final Class<?> target, final Method method) {
+    // We create span and immediately update its name
+    // We do that in order to reuse logic inside updateSpanNames method, which is used externally as
+    // well.
+    Span span = tracer.spanBuilder("jax-rs.request").startSpan();
+    updateSpanNames(span, getCurrentServerSpan(), target, method);
```

Comment:
Have you seen my comment? :) Setting name to builder would lead to almost identical two methods in this class.

Reply:
ah, I missed the comment. I still think better to set the span name directly on span builder instead of calling updateName right afterwards, even if it requires some duplicate code, but not a big deal